morph/react/get-body.js

```diff
@@ -20,81 +22,83 @@ export default ({ state, name }) => {
 
   const maybeTracking = state.track && !state.debug
 
-  const composeAnimation = (state, animation, index) => {
-    if (animation.curve === 'spring') {
-      return composeSpring(animation, index)
-    } else if (state.isReactNative) {
-      return composeTiming(animation, index)
-    }
-  }
+  let maybeAnimated = false
+  let animatedOpen = []
+  let animatedClose = []
+  if (state.isAnimated) {
+    maybeAnimated = true
 
-  const composeSpring = (animation, index) =>
-    `if (props.${animation.scope} !== prev.${animation.scope}) {
-      Animated.spring(this.animatedValue${index}, {
-        toValue: props.${animation.scope} ? 1 : 0,
-        speed: ${animation.speed},
-        bounciness: ${animation.bounciness},
-        delay: ${animation.delay},
-        useNativeDriver: ${canUseNativeDriver(animation)}
-      }).start()
-    }`
-
-  const composeTiming = (animation, index) =>
-    `if (props.${animation.scope} !== prev.${animation.scope}) {
-      Animated.timing(this.animatedValue${index}, {
-        toValue: props.${animation.scope} ? 1 : 0,
-        duration: ${animation.duration},
-        delay: ${animation.delay},
-        useNativeDriver: ${canUseNativeDriver(animation)}
-      }).start()
-    }`
-
-  const composeValues = (state, animation, index) => {
-    if (animation.curve === 'timing' && !state.isReactNative) return
-    return `animatedValue${index} = new Animated.Value(this.props.${
-      animation.scope
-    } ? 1 : 0)
-    `
-  }
+    Object.keys(state.animations).forEach(blockId => {
+      Object.values(state.animations[blockId]).forEach(item => {
+        const { curve, ...configValues } = item.animation.properties
 
-  const maybeAnimated =
-    state.isAnimated || state.hasAnimatedChild
-      ? `${state.animations
-          .map((animation, index) => {
-            return isNewScope(state, animation, index)
-              ? composeValues(
-                  state,
-                  animation,
-                  getScopeIndex(state, animation.scope)
-                )
-              : ''
-          })
-          .join('')}
-      componentDidUpdate(prev) {
-          const { props } = this
-          ${state.animations
-            .map(animation =>
-              composeAnimation(
-                state,
-                animation,
-                getScopeIndex(state, animation.scope)
-              )
+        if (!state.isReactNative && curve !== 'spring') return
+
+        const tag = curve === 'spring' ? 'Spring' : 'Timing'
+        let config = Object.entries(configValues)
+          .map(([key, value]) => `${key}={${value}}`)
+          .join(' ')
+
+        if (curve !== 'spring') {
+          config = `curve="${curve}" ${config}`
```

Comment:
When is this used? I know it's for timing animations, but it's not appearing in the compiled output when I try timing animations.

Reply:
For native. See the check on line 35 that bails if it isn’t RN.